documentation/Makefile

```
@@ -7,7 +7,7 @@ docprereqs:
 	ln -s . sphinx/.out/html/documentation
 
 docpreview: docprereqs
-	@( cd sphinx/.out/html && ! grep FIXME * && python -m SimpleHTTPServer $$((0x$$(echo ${USER} | $(MD5SUM) | awk '{print $$1}' | cut -c1-8)%8000+8000)) )
+	@( cd sphinx/.out/html && ! grep FIXME * && python3 -m http.server $$((0x$$(echo ${USER} | $(MD5SUM) | awk '{print $$1}' | cut -c1-8)%8000+8000)) )
```

```suggestion
	@( cd sphinx/.out/html && ! grep FIXME * && python3 -m http.server 0)
```
This will then print something like 
> Serving HTTP on 0.0.0.0 port 61280 ...

This has two benefits:

1. The makefile will be more readable
2. It guarantees that you will find a free port as the OS will assign one to you

---

The one benefit of the previous scheme was that the port was stable, allowing you to rebuild docs and continue navigating or refresh your current page. It's probably a slightly worse workflow in the average case to generate a random port but much better in the case that the port you get assigned is in use.

Maybe this could be made to use an environment variable to specify a port, defaulting to 0?

---

Ah I didn't think of this... So yaeh I tend to agree that making this optional could be better. I don't have strong feelings one way or another...

---

It may not be worth the effort in the make build system to do anything special, but I could be convinced otherwise.

---

I think that's reasonable - let's try to find agreement on cmake and keep the make stuff as it is.

---

It's unclear to me if this discussion resulted in a "merge this PR" or "change CMake docpreview behavior" ?

---

we already changed the behavior in cmake. I think the consensus was that we want to keep the make behavior the same (so I think this PR should be merged).